src/essentia/essentiamath.h

```diff
@@ -738,8 +738,88 @@ inline Real hz2hz(Real hz){
   return hz;
 }
 
-inline Real hz2cents(Real hz) {
-  return 12 * std::log(hz/440)/std::log(2.) + 69;
+inline Real cents2hz(Real frequencyB, Real cents) {
+  return frequencyB * powf(2.0, cents / 1200.0);
+}
+
+inline Real hz2cents(Real frequencyA, Real frequencyB) {
+  return 1200 * log2(frequencyA / frequencyB);
+}
+
+inline int hz2midi(Real hz, Real tuningFrequency) {
+  return 69 + (int) round(log2(hz / tuningFrequency) * 12);
+}
+
+inline Real midi2hz(int midiNoteNumber, Real tuningFrequency) {
+  return tuningFrequency * powf(2, (midiNoteNumber - 69) / 12.0);
+}
+
+inline std::string note2root(std::string note) {
+    return note.substr(0, note.size()-1);
+}
+
+inline int note2octave(std::string note) {
+    char octaveChar = note.back();
+    return octaveChar - '0';
+}
+
+inline std::string midi2note(int midiNoteNumber) {
+  const std::vector<std::string> ALL_NOTES { "A", "A#", "B", "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#" };
+  int nNotes = ALL_NOTES.size();
+  int CIdx = 3;
+  int diffCIdx = nNotes - CIdx;
+  int noteIdx = midiNoteNumber - 69;
+  int idx = abs(noteIdx) % nNotes;
+  int octave = (CIdx + 1) + floor((noteIdx + diffCIdx) / nNotes);
+  if (noteIdx < 0) {
+    idx = abs(idx - nNotes) % nNotes;
+  }
+  std::string closest_note = ALL_NOTES[idx] + std::to_string(octave);
+  return closest_note;
+}
+
+inline int note2midi(std::string note) {
+  const std::vector<std::string> ALL_NOTES { "A", "A#", "B", "C", "C#", "D", "D#", "E", "F", "F#", "G", "G#" };
+  int octave = note2octave(note);
+  std::string root = note2root(note);
+  int nNotes = ALL_NOTES.size();
+  int CIdx = 3;
+
+  int noteIdx = floor((octave - (CIdx + 1)) * nNotes);
+  int idx = 0;
+  for (int i = 0; i < nNotes; i++) {
+      if (ALL_NOTES[i] == root) {
```

Comment:
Break out of the loop once the matching  note is found.

Reply:
L796 is doing it by index. However it can be replaced by a `break` to make it more explicit. Thanks again to point it out.